### PR TITLE
Bionic upgrade

### PR DESCRIPTION
**Pull request by dswebbthg**
Seemingly only a few changes needed for the migration to bionic, gnupg as a package can't be found so had to preinstall gpg, some changes to the .welcome.sh file to support python3.  Added in a separate commit for the staging-unstable repo as this probably won't get merged.

### CODE REVIEW

**Comment by Kami** [comment]
Thanks for the contribution.

For the time being we want to keep Xenial as a default. It will probably be a while until we can switch a default base image (we will need to ensure Python 3 stuff on Bionic has been running without any issues on large production deployments for a while, etc.).

The changes itself are good though and we should figure out how we can utilize them to allow people to test docker setup using Ubuntu Bionic as a base image.

Perhaps it should be user configurable some how, or maybe we should build another set of test containers with Bionic as a base image?

/cc @armab @warrenvw

**Comment by armab** [comment]
Yes, exactly what @Kami said.

No quick plans in switching the base image, at least until we start supporting `Bionic` officially with all StackStorm Exchange packs being fully py3 compatible. I think we can start seriously thinking about Bionic in Docker around `st2 v3.1+` release which is not that far away per https://docs.stackstorm.com/roadmap.html

We're not eager to keep optional Bionic images or separated repo fork until that happens.
However it's great to have this contribution ready & waiting for the future timing and I'm sure someone else from our users will find this PR helpful.
Thank you for this work @dswebbthg!

File: base/Dockerfile [hunk 1]
```diff
@@ -32,7 +34,7 @@ ENV LANG='en_US.UTF-8' LANGUAGE='en_US:en' LC_ALL='en_US.UTF-8'
 
 # Install StackStorm, but without UI
 RUN if [ "${ST2_VERSION#*dev}" != "${ST2_VERSION}" ]; then \
-    ST2_REPO=unstable; \
+    ST2_REPO=staging-unstable; \
```
Comment (ytjohn) [inline base/Dockerfile:37]: I don't know the difference between unstable and staging-unstable. Is switching to staging-unstable still "valid" a year later?

**Comment by ytjohn** [comment]
As a note, I build all of the containers using bionic and deployed into a kubernetes cluster. Everything seems to be working. One other Dockerfile to consider updating is st2chatops/Dockerfile.


```iff --git a/st2chatops/Dockerfile b/st2chatops/Dockerfile
index 99b6164..7046338 100644
--- a/st2chatops/Dockerfile
+++ b/st2chatops/Dockerfile
@@ -1,4 +1,4 @@
-FROM ubuntu:xenial
+FROM ubuntu:bionic

 ARG DEBIAN_FRONTEND=noninteractive

@@ -31,6 +31,7 @@ RUN apt-get -qq update \
   && apt-get install -y \
     curl \
     locales \
+    gpg \
   && rm -rf /var/lib/apt/lists/* \
   && locale-gen en_US.UTF-8 \
   && update-locale LANG=en_US.UTF-8 LANGUAGE=en_US:en LC_ALL=en_US.UTF-8
@@ -40,7 +41,7 @@ ENV LANG='en_US.UTF-8' LANGUAGE='en_US:en' LC_ALL='en_US.UTF-8'

 # Install st2chatops
 RUN if [ "${ST2_VERSION#*dev}" != "${ST2_VERSION}" ]; then \
-    ST2_REPO=unstable; \
+    ST2_REPO=staging-unstable; \
   else \
     ST2_REPO=stable; \
   fi \```

File: base/Dockerfile (hunk 1 above)
Comment (armab) [reply base/Dockerfile:37]: We'll need `unstable`.
See more info behind packaging repositories: https://github.com/StackStorm/discussions/wiki/Useful-info-for-Release-Managers#packagecloud-repositories

Comment (armab) [inline base/Dockerfile:37]:
```suggestion
    ST2_REPO=unstable; \
```

File: st2web/Dockerfile [hunk 2]
```diff
@@ -37,14 +38,14 @@ ENV LANG='en_US.UTF-8' LANGUAGE='en_US:en' LC_ALL='en_US.UTF-8'
...
-  && echo "deb http://nginx.org/packages/ubuntu/ xenial nginx" > /etc/apt/sources.list.d/nginx.list \
+  && echo "deb http://nginx.org/packages/ubuntu/ bionic nginx" > /etc/apt/sources.list.d/nginx.list \
   && apt-get update \
   && apt-get install -y nginx \
   && rm -f /etc/apt/sources.list.d/nginx.list
 
 # Install StackStorm Web UI
 RUN if [ "${ST2_VERSION#*dev}" != "${ST2_VERSION}" ]; then \
-    ST2_REPO=unstable; \
+    ST2_REPO=staging-unstable; \
```
Comment (armab) [inline st2web/Dockerfile:48]:
```suggestion
    ST2_REPO=unstable; \
```

**Comment by armab** [comment]
@ytjohn Awesome, can you create a PR for st2chatops?

**Comment by ytjohn** [comment]
@armab Thanks. I had created a [downstream PR](https://github.com/dswebbthg/st2-dockerfiles/pull/1/files) since I can't directly modify this one, but your pushes have incorporated them now.

**Comment by armab** [comment]
Cool, I think I'm going to merge this soon.

The plan is to switch from `Ubuntu:xenial` to `Ubuntu:bionic` for the K8s Dockerfiles in next Helm chart version release.
https://github.com/StackStorm/stackstorm-ha/blob/master/Chart.yaml#L3
Old `3.2dev` Dockerfiles are based on old `Xenial`
The new `3.3dev` Dokerfiles will be based on new `Bionic`
This means StackStorm installation will be py3-based only.

**Pull request merged**